mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/ui/EditProfileActivity.java

```diff
@@ -268,6 +270,23 @@ public void hideKeyboard() {
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
                                            @NonNull int[] grantResults) {
         switch (requestCode) {
+            case REQUEST_CAMERA: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
+                    // camera-related task you need to do.
+                    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                    startActivityForResult(intent, REQUEST_CAMERA);
+
+                } else {
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
+                    Toaster.showToast(this,
+                            Constants.NEED_CAMERA_PERMISSION_TO_CLICK_PROFILE_PICTURE);
```

Comment:
@ReCodee For any user-visible message/text we should use a string resource object as we can easily provide a multi-language alternative for them when needed. For any internal workings of the application we can use `Constants` as the above use case doesn't apply to them.

I just saw this hasn't been followed by the _need storage message_ text also and possibly at many other places too. A separate issue can address this, but yeah we can avoid this for any new additions

Reply:
Thank you, I'll keep this in mind from now on. Btw can I make an issue to resolve this matter?

Reply:
Yeah, if this problem persists at many places, a pull request can be addressed to solving this